ui/component/downloadProgress/view.jsx

```diff
@@ -0,0 +1,206 @@
+// @flow
+import React, { useState } from 'react';
+import { shell } from 'electron';
+import Icon from 'component/common/icon';
+import * as ICONS from 'constants/icons';
+
+type Props = {
+  downloadList: any,
+  stopDownload: (outpoint: string, sd_hash: string) => void,
+};
+
+function DownloadProgress(props: Props) {
+  const [isShow, setIsShow] = useState(true);
+  const [cancelHash] = useState({});
+
+  const handleCancel = (hash, value) => {
+    cancelHash[hash] = value;
+  };
+  // return (
+  //   <div className="download--header">
+  //     <button className="download--top-close-button" onClick={() => setIsShow(false)}>
+  //       <div />
+  //     </button>
+  //     <DownloadState
+  //       fileName="being-offended-is-a-bad-trait.mp4"
+  //       title="Being offended is a bad trait"
+  //       releaseTime={'9/27/2021'}
+  //       writtenBytes={100000}
+  //       totalBytes={500000}
+  //       addedOn={1632388934}
+  //       directory={''}
+  //       stopDownload={props.stopDownload}
+  //       outpoint={'asdfasdfsa'}
+  //       sd_hash={'asdfsadfsadfsd'}
+  //       isCancel
+  //       handleCancel={handleCancel}
+  //     />
+  //   </div>
+  // );
+
+  // console.log('DownloadProgress', props.downloadList);
+  if (!props.downloadList) return null;
+
+  const downloadList = [];
+  Object.keys(props.downloadList).map((key) => {
+    const item = props.downloadList[key];
```

Comment:
something weird happens here that crashes the app when I play a video.
"cannot read status of null"
redux devtools (which has a bug that sometimes doesn't load unless you `yarn dev` and wait for the app to finish before opening devtools) tells me initial state byOutpoint is { <outpoint>: null } until file_list fetch comes back.